storage/replica_test.go

```diff
@@ -2675,6 +2683,137 @@ func TestRangeDanglingMetaIntent(t *testing.T) {
 	}
 }
 
+// TestRangeLookupUseReverseScan verifies the correctness of the results which are retrived
+// from RangeLookup by using ReverseScan.
+func TestRangeLookupUseReverseScan(t *testing.T) {
+	defer leaktest.AfterTest(t)
+	tc := testContext{}
+	tc.Start(t)
+	defer tc.Stop()
+
+	// Test ranges: ["a","c"), ["c","f"), ["f","h") and ["h","y").
+	testRanges := []proto.RangeDescriptor{
+		{RangeID: 2, StartKey: proto.Key("a"), EndKey: proto.Key("c")},
+		{RangeID: 3, StartKey: proto.Key("c"), EndKey: proto.Key("f")},
+		{RangeID: 4, StartKey: proto.Key("f"), EndKey: proto.Key("h")},
+		{RangeID: 5, StartKey: proto.Key("h"), EndKey: proto.Key("y")},
+	}
+	// The range ["f","h") has dangling intent in meta2.
+	withIntentRangeIndex := 2
+
+	testCases1 := []struct {
+		key      string
+		expected proto.RangeDescriptor
+	}{
+		// testRanges[0|1] exists, use the key in the midlle or end of
```

Comment:
// For testRanges{0,1} there is no intent. A key in the middle and the end key should both give us the range itself.